pkg/lockfile/parse-pdm-lock.go

```diff
@@ -0,0 +1,80 @@
+package lockfile
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/BurntSushi/toml"
+)
+
+type PdmLockPackage struct {
+	Name     string   `toml:"name"`
+	Version  string   `toml:"version"`
+	Groups   []string `toml:"groups"`
```

Comment:
just noting for others: it looks like pdm only _very_ recently started recording dependency groups at the package level in its lockfiles - I couldn't actually find any locks searching GitHub, but the PRs on pdm support this feature as being added